### PR TITLE
Update automated analysis

### PR DESCRIPTION
**Pull request by IsaackMwenda**
This updates the standard automated analysis logic (engagement, demographic and theme distributions) to use Analysis Util in Core as is on IBTCI.
Will update the other project-specific code in the next pr.

### CODE REVIEW

File: automated_analysis.py [hunk 1]
```diff
@@ -222,197 +211,23 @@
...
-                last_row_episode = episode
+        theme_distributions.export_theme_distributions_csv(
+            individuals, CONSENT_WITHDRAWN_KEY,
+            coding_plans_to_analysis_configurations(PipelineConfiguration.RQA_CODING_PLANS),
+            coding_plans_to_analysis_configurations(PipelineConfiguration.SURVEY_CODING_PLANS),
+            f
+        )
 
     # Export raw messages labelled with Meta impact, gratitude and about conversation programmatically known as impact/success story
     log.info("Exporting success story raw messages for each episode...")
```
Comment (as2388) [inline automated_analysis.py:233]: It should be possible to use sample_messages in Core to do this bit. If not, I hope we can update the implementation of sample_messages so we can.

File: automated_analysis.py [hunk 2]
```diff
@@ -68,6 +69,27 @@
...
+    log.info("Computing engagement counts...")
+    with open(f"{automated_analysis_output_dir}/engagement_counts.csv", "w") as f:
+        engagement_counts.export_engagement_counts_csv(
+            messages, individuals, CONSENT_WITHDRAWN_KEY,
+            coding_plans_to_analysis_configurations(PipelineConfiguration.RQA_CODING_PLANS),
+            f
+        )
+
     # Compute the number of messages, individuals, and relevant messages per episode and overall.
     log.info("Computing the per-episode and per-season engagement counts...")
```
Comment (as2388) [inline automated_analysis.py:94]: You should be able to remove this given the engagement counts export above

File: automated_analysis.py [hunk 3]
```diff
@@ -109,46 +131,13 @@
...
-
-        for row in repeat_participations.values():
-            writer.writerow(row)
+        repeat_participations.export_repeat_participations_csv(
+            individuals, CONSENT_WITHDRAWN_KEY,
+            coding_plans_to_analysis_configurations(PipelineConfiguration.RQA_CODING_PLANS),
+            f
+        )
 
     log.info(f'Computing repeat and new participation per show ...')
```
Comment (as2388) [inline automated_analysis.py:142]: How widely is this being used? This could be useful to standardise at some point.

**Review by as2388 — APPROVED** [review]

File: automated_analysis.py (hunk 1 above)
Comment (IsaackMwenda) [reply automated_analysis.py:233]: Thanks forgot this here.

File: automated_analysis.py (hunk 2 above)
Comment (IsaackMwenda) [reply automated_analysis.py:94]: Thanks forgot this here. Removed.

File: automated_analysis.py (hunk 3 above)
Comment (IsaackMwenda) [reply automated_analysis.py:142]: Mainly used in Wusc and Imaqal but its a relevant metric, will standardize and move it to core.

File: automated_analysis.py (hunk 1 above)
Comment (IsaackMwenda) [reply automated_analysis.py:184]: updated to use sample_messagea in Core.

**Pull request merged**